tests/Dfc.CourseDirectory.WebV2.Tests/FeatureTests/Courses/DeleteCourseRunTests.cs

```diff
@@ -0,0 +1,534 @@
+                HttpMethod.Get,
+                $"/courses/delete?courseId={courseId}&courseRunId={courseRunId}&returnUrl=%2Fcourses");
+
+            await User.AsTestUser(userType, anotherProviderId);
+
+            // Act
+            var response = await HttpClient.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.Forbidden, response.StatusCode);
+        }
+
+        [Fact]
+        public async Task Get_ValidRequest_RendersExpectedOutput()
+        {
+            // Arrange
+            var providerId = await TestData.CreateProvider();
+
+            var courseId = await TestData.CreateCourse(
+                providerId,
+                qualificationCourseTitle: "Maths",
+                createdBy: User.ToUserInfo());
+
+            var courseRunId = await GetCourseRunIdForCourse(courseId);
+
+            var request = new HttpRequestMessage(
+                HttpMethod.Get,
+                $"/courses/delete?courseId={courseId}&courseRunId={courseRunId}&returnUrl=%2Fcourses");
+
+            await User.AsProviderUser(providerId, ProviderType.FE);
+
+            // Act
+            var response = await HttpClient.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            var doc = await response.GetDocument();
+            Assert.Equal("Maths", doc.GetSummaryListValueWithKey("Course name"));
+        }
+
+        [Fact]
+        public async Task Get_ValidRequestCourseRunWithVenue_RendersLocationRow()
+        {
+            // Arrange
+            var providerId = await TestData.CreateProvider();
+
+            var venueId = await TestData.CreateVenue(providerId, venueName: "Test Venue");
+
+            var courseId = await TestData.CreateCourse(
+                providerId,
+                qualificationCourseTitle: "Maths",
+                createdBy: User.ToUserInfo(),
+                configureCourseRuns: b => b.WithCourseRun(
+                    CourseDeliveryMode.ClassroomBased,
+                    CourseStudyMode.FullTime,
+                    CourseAttendancePattern.Daytime,
+                    venueId: venueId));
+
+            var courseRunId = await GetCourseRunIdForCourse(courseId);
+
+            var request = new HttpRequestMessage(
+                HttpMethod.Get,
+                $"/courses/delete?courseId={courseId}&courseRunId={courseRunId}&returnUrl=%2Fcourses");
+
+            await User.AsProviderUser(providerId, ProviderType.FE);
+
+            // Act
+            var response = await HttpClient.SendAsync(request);
+
+            // Assert
+            Assert.Equal(HttpStatusCode.OK, response.StatusCode);
+
+            var doc = await response.GetDocument();
+            Assert.Equal("Test Venue", doc.GetSummaryListValueWithKey("Course location"));
+        }
+
+        [Fact]
+        public async Task Get_ValidRequestCourseRunWithNoVenue_DoesNotRenderLocationRow()
```

Comment:
It doesn't jump out at me what the key change in data is compared to the above test that results in rendering vs not rendering. Maybe a code comment on the relevant item(s) would suffice.

Reply:
Added explicit `venueId: null` to the `CreateCourse` method